### PR TITLE
fix: add client_max_body_size and gzip on to nginx

### PR DESCRIPTION
**Pull request by varl**
Align closer to the best practices of DHIS2 deployments: https://docs.dhis2.org/master/en/implementer/html/dhis2_implementation_guide_full.html#install_basic_nginx_setup

### CODE REVIEW

File: cluster/config/nginx.conf [hunk 1]
```diff
@@ -9,13 +9,21 @@ http {
...
+    "text/css;charset=utf-8" text/css
+    "text/plain;charset=utf-8" text/plain;
+
   server {
     listen 80;
     port_in_redirect off;
 
     root /data/apps;
 
-    rewrite ^/$ http://$http_host/dhis-web-dashboard redirect;
```
Comment (varl) [inline cluster/config/nginx.conf]: Not sure why we would need this? On the contrary I keep getting redirected to this URL which doesn't exist. 🤔

**Comment by amcgee** [comment]
@ismay since your experienced behavior changed I cleared your review, would you mind testing again?  How are you testing it?

**Comment by ismay** [comment]
> @ismay since your experienced behavior changed I cleared your review, would you mind testing again? How are you testing it?

I'm changing the config directly in .cache/d2. And restarting the containers.

**Comment by ismay** [comment]
Just retested to make sure. Still the same (aka 413 error) results for me.

**Comment by varl** [comment]
It's strange. The default body size for a request in nginx is 1 MB. So 1.2 MB _should_ fail, but setting it to 10 MB should allow it to go through...

**Comment by ismay** [comment]
> It's strange. The default body size for a request in nginx is 1 MB. So 1.2 MB should fail, but setting it to 10 MB should allow it to go through...

Yeah. Do you guys have the same? I'm testing with the import/export app. We could also just merge this, as using the same nginx config as play seems good anyway.

**Comment by varl** [comment]
I've verified that it fixes the problem for me.

Config loaded in the container for the gateway:

```
varl@dhis-tree:~/cli$ docker exec -it d2-cluster-dev_gateway_1 bash
bash-4.4# cat /etc/nginx/nginx.conf

worker_processes 1;

events {
  worker_connections 1024;
}

http {
  upstream core {
    server core:8080 max_fails=3 fail_timeout=60s;
  }

  gzip on; # Enables compression, incl Web API content-types
  gzip_types
    "application/json;charset=utf-8" application/json
    "application/javascript;charset=utf-8" application/javascript text/javascript
    "application/xml;charset=utf-8" application/xml text/xml
    "text/css;charset=utf-8" text/css
    "text/plain;charset=utf-8" text/plain;

  server {
    listen 80;
    port_in_redirect off;

    root /data/apps;

    client_max_body_size 10m;

    location / {
      include mime.types;
      try_files $uri $uri/ $uri/index.html @core;
    }

    # location /screenshot {
    #   proxy_set_header Host $http_host;
    #   proxy_set_header X-Real-IP $remote_addr;
    #   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    #   proxy_pass http://screenshot:9000;
    # }

    location @core {
      proxy_set_header Host $http_host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_pass http://core;
    }
  }
}
```

Used the GML file from this issue: https://jira.dhis2.org/browse/DHIS2-5703 named `Valid_SierraLeone_OU.gml`.

Before I applied the patch I got a `413` response code.

After applying this patch I got a `200` response code.

**Comment by ismay** [comment]
Nice. Yeah I think for me it's probably just something weird with my setup that's causing it to not pick up the config changes.

**Comment by varl** [comment]
Merging fix for the issue. Let me know if I should add the rewrite rule back (I don't see why it's needed).

**Comment by amcgee** [comment]
Great!  Nope, seems fine to remove it.

**Pull request merged**